File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum = nums[0]
        sum=0

        for i in range(len(nums)):
            if sum<0:
                sum=0
                
            sum = sum + nums[i]
            max_sum=max(sum,max_sum)

        return max_sum

        

File: 0053-maximum-subarray/0053-maximum-subarray.c
int maxSubArray(int* nums, int numsSize) {
    int maxsum = nums[0];
    int sum=0;
    int i;

    for(i=0;i<numsSize;i++) {
        sum = sum + nums[i];
        if(sum>maxsum) {
            maxsum = sum;
        }
        if(sum<0) {
            sum=0;
        }
    }
    return maxsum;
}